packages/console/src/pages/EnterpriseSso/SsoConnectorLogo/index.tsx

```diff
@@ -23,9 +23,11 @@ const pickLogoForCurrentTheme = (
   branding: SsoConnectorWithProviderConfig['branding']
 ): string => {
   // Need to use `||` here since `??` operator can not avoid empty strings.
-  // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-  const configuredLogo = isDarkMode ? branding.darkLogo : branding.logo || branding.darkLogo;
-  const builtInLogo = isDarkMode ? logoDark : logo || logoDark;
+  // Since `logo` and `darkLogo` are both optional, when it is dark mode and `darkLogo` is not configured, should fallback to `logo`.
+  const configuredLogo =
+    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+    (isDarkMode ? branding.darkLogo : branding.logo) || branding.darkLogo || branding.logo;
+  const builtInLogo = (isDarkMode ? logoDark : logo) || logoDark || logo;
```

Comment:
This is still confusing, at least I do not understand what should be the correct logic. May need to discuss it offline and sort it out

Reply:
Can tell the functionality directly from the test cases.